src/TimeInput.js

```diff
@@ -48,26 +53,49 @@ class TimeInput extends React.Component {
 
   handleCancel = () => this.setState({ open: false, newValue: null })
 
+  getFormattedValue = () => {
+    const { mode, defaultValue } = this.props
+    const { value, hasChanged } = this.state
+
+    const is12h = mode === '12h'
+
+    // hasOwnProperty is important here, we want `null` and `undefined` to display different things.
+    if (this.props.hasOwnProperty('defaultValue') && !hasChanged) {
+      // Empty string if defaultValue is null
+      return defaultValue == null ? '' : defaultValue
+    }
+
+    // Allow a null/undefined value for controlled inputs
+    if (this.props.hasOwnProperty('value') && this.props.value == null) {
```

Comment:
This is a different use case, here we are checking if the prop is passed in at all, and only then do we allow for its value to be undefined/null. Without this someone could pass in `value={undefined}` and the component would break. On the other side if someone were to omit the value prop totally `this.props.value == null` would be true because the property is truly undefined. I think this just saves the end user from experiencing bugs if they, for example, use an uninitialized variable with a controlled component.